### PR TITLE
Provenance for CLI

### PR DESCRIPTION
**Pull request by datosh**
### Proposed change(s)

- Use [slsa-generator](https://github.com/slsa-framework/slsa-github-generator) and [slsa-verifier](https://github.com/slsa-framework/slsa-verifier) to generate provenance: ["a positive attestation to a software artifact coming from [our] repository"](https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md#benefits-of-provenance), and verify it as a check and documentation.
- The provenance is signed with cosign (internally in the generator). It is not possible to provide your own keys to the slsa-generator step ([yet](https://github.com/slsa-framework/slsa-github-generator/issues/326)). The GitHub OIDC flow is used (see `permissions: id-token: write`). I think this is fine for us, since the pipeline signs it has run the build. If someone controls the pipeline they could swap our provided key. This would only be a real advantage if we choose to run self-hosted runners.
- Refactored build CLI workflow to run steps in parallel:
  - [1st successful run with provenance (15min)](https://github.com/edgelesssys/constellation/actions/runs/3541919788)
  - [final run (5m33s)](https://github.com/edgelesssys/constellation/actions/runs/3542631038)

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Link to Milestone

### CODE REVIEW

**Comment by datosh** [comment]
After chatting with @3u13r & @malt3: I extended provenance to include SBOM. [Pipeline run.](https://github.com/edgelesssys/constellation/actions/runs/3547268867)

File: docs/docs/workflows/verify-cli.md [hunk 1]
```diff
@@ -40,6 +40,30 @@ tlog entry verified with uuid: afaba7f6635b3e058888692841848e5514357315be9528474
 Verified OK
 ```
 
+## Verify the provenance
+
+Provenance is a positive attestation that a software artifact was produced by a specific repository and build system invocation. For more information on provenance visit [slsa.dev](https://slsa.dev/provenance/v0.2).
```
Comment (3u13r) [inline docs/docs/workflows/verify-cli.md:45]: Is "positive attestation" a known phrase? Otherwise I'd rephrase to something like:
```suggestion
Provenance attests that a software artifact was produced by a specific repository and build system invocation. For more information on provenance visit [slsa.dev](https://slsa.dev/provenance/v0.2).
```

**Review by 3u13r — APPROVED** [review]
small nit, otherwise LGTM
